pontoon/sync/tests/formats/test_ftl.py

```diff
@@ -0,0 +1,291 @@
+
+NoCommentsorSources = Translated No Comments or Sources
+"""
+
+
+class FTLTests(FormatTestsMixin, TestCase):
+    parse = staticmethod(ftl.parse)
+    supports_keys = False
+    supports_source = False
+    supports_source_string = False
+
+    def setUp(self):
+        super(FTLTests, self).setUp()
+        self.tempdir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        super(FTLTests, self).tearDown()
+        shutil.rmtree(self.tempdir)
+
+    def key(self, source_string):
+        """FTL keys can't contain spaces."""
+        return super(FTLTests, self).key(source_string).replace(' ', '')
+
+    def test_parse_basic(self):
+        input_string = BASE_FTL_FILE
+        translation_index = 0
+        path, resource = self.parse_string(input_string)
+
+        assert_attributes_equal(
+            resource.translations[translation_index],
+            comments=['Sample comment'],
+            key=self.key('Source String'),
+            strings={None: 'SourceString = Translated String\n'},
+            fuzzy=False,
+            order=translation_index,
+        )
+
+    def test_parse_multiple_comments(self):
+        input_string = BASE_FTL_FILE
+        translation_index = 1
+        path, resource = self.parse_string(input_string)
+
+        assert_attributes_equal(
+            resource.translations[translation_index],
+            comments=['First comment\nSecond comment'],
+            source=[],
+            key=self.key('Multiple Comments'),
+            strings={None: 'MultipleComments = Translated Multiple Comments\n'},
+            fuzzy=False,
+            order=translation_index,
+        )
+
+    def test_parse_no_comments_no_sources(self):
+        input_string = BASE_FTL_FILE
+        translation_index = 2
+        path, resource = self.parse_string(input_string)
+
+        assert_attributes_equal(
+            resource.translations[translation_index],
+            comments=[],
+            source=[],
+            key=self.key('No Comments or Sources'),
+            strings={None: 'NoCommentsorSources = Translated No Comments or Sources\n'},
+            fuzzy=False,
+            order=translation_index,
+        )
+
+    def test_save_basic(self):
+        input_string = dedent("""
+            SourceString = Source String
+        """)
+        expected_translation = 'SourceString = New Translated String'
+        expected_string = dedent("""
+            {expected_translation}
+        """.format(
+            expected_translation=expected_translation
+        ))
+
+        self.run_save_basic(
+            input_string,
+            expected_string,
+            source_string=input_string,
+            expected_translation=expected_translation,
+        )
+
+    def test_save_remove(self):
+        """Deleting strings removes them completely from the FTL file."""
+        input_string = dedent("""
+            Source-String = Source String
+        """)
+        expected_string = dedent("""""")
```

Comment:
These 2 strings can easily fit on a single line, removing the need for triple quotes and `dedent`.